File: tests/testdata/codegen/deferred-outputs-pp/python/first.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import pulumi_random as random

class FirstArgs(TypedDict, total=False):
    passwordLength: Input[float]

class First(pulumi.ComponentResource):
    def __init__(self, name: str, args: FirstArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:First", name, args, opts)

        random_pet = random.RandomPet(f"{name}-randomPet", opts = pulumi.ResourceOptions(parent=self))

        random_password = random.RandomPassword(f"{name}-randomPassword", length=args["passwordLength"],
        opts = pulumi.ResourceOptions(parent=self))

        self.petName = random_pet.id
        self.password = random_password.result
        self.register_outputs({
            'petName': random_pet.id, 
            'password': random_password.result
        })

File: tests/testdata/codegen/deferred-outputs-pp/go/first.go
package main

import (
	"fmt"

	"github.com/pulumi/pulumi-random/sdk/v4/go/random"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

type FirstArgs struct {
	PasswordLength pulumi.Float64Input
}

type First struct {
	pulumi.ResourceState
	PetName  pulumi.AnyOutput
	Password pulumi.AnyOutput
}

func NewFirst(
	ctx *pulumi.Context,
	name string,
	args *FirstArgs,
	opts ...pulumi.ResourceOption,
) (*First, error) {
	var componentResource First
	err := ctx.RegisterComponentResource("components:index:First", name, &componentResource, opts...)
	if err != nil {
		return nil, err
	}
	randomPet, err := random.NewRandomPet(ctx, fmt.Sprintf("%s-randomPet", name), nil, pulumi.Parent(&componentResource))
	if err != nil {
		return nil, err
	}
	randomPassword, err := random.NewRandomPassword(ctx, fmt.Sprintf("%s-randomPassword", name), &random.RandomPasswordArgs{
		Length: args.PasswordLength,
	}, pulumi.Parent(&componentResource))
	if err != nil {
		return nil, err
	}
	err = ctx.RegisterResourceOutputs(&componentResource, pulumi.Map{
		"petName":  randomPet.ID(),
		"password": randomPassword.Result,
	})
	if err != nil {
		return nil, err
	}
	componentResource.PetName = randomPet
	componentResource.Password = randomPassword.Result
	return &componentResource, nil
}
